src/EscapeAnalysis.jl

```diff
@@ -125,6 +329,24 @@ x::EscapeLattice == y::EscapeLattice = begin
 end
 
 x::EscapeLattice ⊑ y::EscapeLattice = begin
+    xf, yf = x.FieldSets, y.FieldSets
+    if isa(xf, Bool)
+        xf && yf !== true && return false
+    else
+        if isa(yf, Bool)
+            yf === false && return false
+        else
+            xf, yf = xf::FieldsInfo, yf::FieldsInfo
+            xn, yn = length(xf), length(yf)
+            xn > yn && return false
+            for i in 1:xn
+                xf[i] ⊆ yf[i] || return false
+            end
+        end
+    end
+    return x ⊑ₑ y
+end
+x::EscapeLattice ⊑ₑ y::EscapeLattice = begin # partial order excluding the `FieldSets` order
```

Comment:
Can you explain why we need both partial orders?

Reply:
The reason was that `FieldSets` isn't a proper "escape property" and thus should not participate in lattice order calculation. Well, now I feel that design was wrong.